### PR TITLE
Refactoring of the dialog base visuals, new button structure.

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR changes the implementation of the `CommonDialogWindowBaseContent` class to be a `ContentControl` and simplifies how the control is to be used when implementing dialogs. Because of the new approach, and the use of the new class `BindableList<T>` there's no longer a need to have the main UI for a dialog as a resource, bindings just work in a more natural fashion.

This PR also introduces the new styles for buttons in the dialogs, and the `ButtonTemplateSelector` class to choose between the two available button sizes. 

This PR also marks the first introduction of the CommonResources.xaml resource dictionary where all of the common resources, such as font sizes, `TextBlock` styles, etc... will be stored.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.Utils/BindableList.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    public class BindableList<T> : IList where T : FrameworkElement
+    {
+        private readonly IList<T> _storage = new ObservableCollection<T>();
+        private readonly DependencyObject _dataContextSource;
+
+        public IList<T> Collection => _storage;
+
+        #region IList
+
+        bool IList.IsReadOnly => _storage.IsReadOnly;
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.Utils/BindableList.cs]: Should these all have " /// <inheritdoc />" comments on them?

File: GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+    /// <summary>
+    /// This class implements the <seealso cref="DataTemplateSelector"/> interface to choose the right
+    /// button style (with the right size) depending on the contents of the button.
+    /// </summary>
+    public class ButtonTemplateSelector : DataTemplateSelector
+    {
+        /// <summary>
+        /// Choose between the Wide and Standard sizes for a button depending on the information on 
+        /// the button fino.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs]: spelling? fino?

File: GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        /// </summary>
+        public override DataTemplate SelectTemplate(object item, DependencyObject container)
+        {
+            var buttonInfo = (DialogButtonInfo)item;
+            var element = (FrameworkElement)container;
+
+            if (buttonInfo.Caption.Length > 6)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs]: Can you either make this a const or comment where 6 comes from?

File: GoogleCloudExtension/GoogleCloudExtension.Utils/BindableList.cs (hunk 1 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.Utils/BindableList.cs]: They're just implementing `IList`.

File: GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs (hunk 2 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs]: Done, it was meant to be _info_.

File: GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs (hunk 3 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/Theming/ButtonTemplateSelector.cs]: Added comment, the number is more or less arbitrary, mostly to optimize the looks.

**Pull request merged**